main.go

```diff
@@ -43,42 +47,71 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
-func main() {
-	var metricsAddr string
-	var enableLeaderElection bool
-	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
+var flags = struct {
+	EnableLeaderElection bool
+	MetricsAddr          string
+}{}
+
+func initFlags() {
+	flag.BoolVar(&flags.EnableLeaderElection, "enable-leader-election", false, "Enable leader election for controller manager.")
+	flag.StringVar(&flags.MetricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.Parse()
 
+	var errors []string
+
+	// Flag validation goes here.
+	//
+	//if flags.MyFlag == "" {
+	//	errors = append(errors, "--my-flag must be not empty")
+	//}
+
+	if len(errors) > 1 {
+		fmt.Fprintf(os.Stderr, "%s\n", strings.Join(errors, "\n"))
+		os.Exit(2)
+	}
+}
```

Comment:
This is my idea of using flags here. I think it's pretty robust. LMK what you think (if you care).